src/index.js

```diff
@@ -1,57 +1,56 @@
-const fetch = require("node-fetch")
 const {getColor, getRandomColor} = require("./colors")
-
 class Hue {
-    constructor(ip, username) {
-        this.api = `http://${ip}/api/${username}`
+    constructor(hueClient) {
+        this.client = hueClient
     }
 
-    readLight = async id => {
-        const response = await fetch(`${this.api}/lights/${id}`, {
-            method: "GET",
-        })
-
-        const json = await response.json()
-        const light = {id, ...json}
-
-        return light
+    getLightInfo = async id => {
+        return this.client.readLight(id)
     }
 
-    readLights = async () => {
-        const response = await fetch(`${this.api}/lights`, {method: "GET"})
-        const json = await response.json()
-
-        const lights = Object.entries(json).map(([id, light]) => {
-            return {id, ...light}
-        })
+    getLightsInfo = async () => {
+        return this.client.readLights(this.client)
+    }
 
-        return lights
+    turnOnLight = id => {
+        this.client.updateLight(id, {on: true}, )
     }
 
-    updateLight = (id, state) => {
-        fetch(`${this.api}/lights/${id}/state`, {
-            method: "PUT",
-            body: JSON.stringify(state),
-        })
+    turnOffLight = id => {
+        this.client.updateLight(id, {on: false}, )
+    }
+    
+    blinkLight = async id => {
```

Comment:
Let's consider making the interval configurable, and allowing the user to specify how many times (`count`) to blink the light.

```javascript
blinkLight = async (id, interval = 500, count = 1)
```